### DISCUSSION TITLE
Is it available to get dll of previous or old version Magick.Native (Magick.Native-Q16-x64.dll etc) other than dropbox?

### DISCUSSION THREAD

**Discussion by t246246** [dlemstra/Magick.NET#679] [General] [answered]
I am trying to build silghtly older Magick.NET version, 7.19.0.1.
I successed last week and am trying to do the same step in new environment.
When I run src/Magick.Netive/install.cmd, I got http 404 error:

```
+     Invoke-WebRequest $url -Outfile "$temp\Magick.Native.$version.nup ...
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest]、WebExce
    ption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
```

I understand it is on dropbox and cannot keep so many older versions.  (I just confirmed I can get them for 7.20.0.1 as the time of writing.) but if I want to build older version in new enviromnent, how to get these files?

For my purpose these files are temporarily needed, but next build step (Copy-Item) requires these files, Magick.Native-Q16-x64.dll, Magick.Native-Q16-x64.gz etc.  Although these have to rebuild in my environment again, as I need to slight modify to ImageMagick source.

**Comment by dlemstra** [comment, marked as answer]
Why do you need to copy these files when you are modifying the ImageMagick source and creating your own version of the Native library?

**Reply by t246246** [reply]
Thank you for your reply.  Yes it is very natural question.

Basically I run the script comes with Magick.NET and Magick.Native one by one:

Currently I run the follwoing batch firstly:

```
cd /d %~dp0

set MagickNET=.\Magick.NET-7.20.0.1
set MagickNative=.\Magick.Native-5202ccb01f6df67d777a14fe465b627796714a49
set OPATH=%PATH%

cd /d %~dp0
cd %MagickNET%\build\windows
if not exist ..\..\tools\windows\nuget.exe call install.dependencies.cmd

cd /d %~dp0
cd %MagickNET%\src\Magick.Native
if not exist libraries call install.cmd

cd /d %~dp0
cd %MagickNative%\build\dotnet\windows
if not exist ..\..\..\tools\windows\nuget.exe powershell ./install.dependencies.ps1

cd ..\..\..\src\ImageMagick
if not exist libraries call checkout.cmd
```

then edit the ImageMagick source, then run next batch that will call build.ImageMagick.bat, build.Native.bat etc etc.

The first batch failed when download failed.  So I am afraid I am not running these commands in proper order, or doing unnecessary jobs.

I will try to analyze your batch and powershell, but I change my question now:

Are there good way to build both Magick.Native and Magick.NET without using install.cmd?

**Reply by t246246** [reply]
I checked I can build it without install.cmd at first.
Simply I need to crate emty folder 'libraries' at src/Magick.Native in Magick.NET project.

So close this discussion.
